src/plugins/data/public/data_sources/datasource/types.ts

```diff
@@ -3,49 +3,79 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { DataSource } from './datasource';
+
 /**
  * @experimental These interfaces are experimental and might change in future releases.
  */
 
-import { IndexPatternsService } from '../../index_patterns';
-import { DataSourceType } from '../datasource_services';
-
 export interface IndexPatternOption {
   title: string;
   id: string;
 }
 
-export interface IDataSourceMetaData {
+export interface IDataSourceGroup {
   name: string;
 }
 
-export interface IDataSourceGroup {
+export interface DataSetWithDataSource<T = []> {
+  ds: DataSource;
+  list: T;
+}
+
+export interface IDataSetParams<T = {}> {
+  query: T;
+}
+
+export interface IDataSourceQueryParams<T = {}> {
+  query: T;
+}
+
+export interface IDataSourceQueryResult<T = {}> {
```

Comment:
why are the generics defaulting to `{}`?

Reply:
For these interfaces basically there's no concrete implementations needed at this point so just providing a default object so elsewhere it doesn't need to specify a type which it doesn't really need.